### PR TITLE
feat: w2d list item

### PR DESCRIPTION
**Pull request by m6jones**
Math, Science and so on are the course names.
image

https://rally1.rallydev.com/#/357252966636d/custom/367300408400?detail=%2Fuserstory%2F514438875868

Get the data required for each activity..

### CODE REVIEW

File: components/activity/description/custom/DescriptionMixin.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	}
+
+	render() {
+		if (this.skeleton) return html`Description<div class="d2l-activity-description-skeleton-extend-skeleton-width"></div>`;
+		return html`
+			<span class="d2l-skeletize">${this._description ? this._description : nothing}</span>
+		`;
+	}
+
+	get _loaded() {
```
Comment (lloydwaddell) [inline components/activity/description/custom/DescriptionMixin.js:31]: Not for this story, but curious if this.skeleton could end up being abstracted even more and just be available to all components by default. Then they would only need to handle the render portion

File: components/activity/description/custom/d2l-activity-description-discussion.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { customHypermediaElement } from '@brightspace-hmc/foundation-engine/framework/lit/hypermedia-components.js';
+import { DescriptionMixin } from './DescriptionMixin.js';
+import { observableTypes } from '@brightspace-hmc/foundation-engine/framework/lit/HypermediaStateMixin.js';
+
+const rels = Object.freeze({
+	description: 'https://discussions.api.brightspace.com/rels/description',
+	topic: 'https://discussions.api.brightspace.com/rels/topic'
+});
+
+export class ActivityDiscussionAssignment extends DescriptionMixin {
```
Comment (lloydwaddell) [inline components/activity/description/custom/d2l-activity-description-discussion.js:10]: nit: should this be ActivityDescriptionDiscussion

File: features/workToDo/d2l-w2d-list-item.js [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+			'd2l-focusing': this._focusingPrimaryAction,
+		};
+
+		return this._renderListItem({
+			illustration: html`<d2l-activity-icon href="${this.href}" .token="${this.token}" class="${classMap(iconClasses)}"></d2l-activity-icon>`,
+			content: html`${guard([this.href, this.token], () => html`
+				<d2l-list-item-content>
+					<d2l-activity-name class="d2l-w2d-list-item-name" href="${this.href}" .token="${this.token}"></d2l-activity-name>
+					<div slot="secondary">
+						${this._isCourse ? html`Course` : html`<d2l-activity-name-course href="${this.href}" .token="${this.token}"></d2l-activity-name-course>`}
```
Comment (lloydwaddell) [inline features/workToDo/d2l-w2d-list-item.js:89]: nit: _isCourse isn't obvious where it is coming from, could it be added to the constructor / properties?

File: features/workToDo/d2l-w2d-list-item.js [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+			content: html`
+				<d2l-list-item-content skeleton>
+					<d2l-activity-name class="d2l-w2d-list-item-name" skeleton></d2l-activity-name>
+				</d2l-list-item-content>
+			`
+		});
+	}
+}
+
+class W2DListItem extends W2DListItemMixin {}
```
Comment (lloydwaddell) [inline features/workToDo/d2l-w2d-list-item.js:113]: nit: so many classes :D lol (sorry had to)

**Review by lloydwaddell — APPROVED** [review]

File: components/activity/description/custom/DescriptionMixin.js (hunk 1 above)
Comment (m6jones) [reply components/activity/description/custom/DescriptionMixin.js:31]: I might just change _loaded to skeleton?

**Pull request merged**